### PR TITLE
gh-102828: emit deprecation warning for onerror arg to shutil.rmtree

### PR DESCRIPTION
**Pull request by iritkatriel**
I've asked on discord and was told that we normally emit the deprecation warning as soon as something is deprecated. 

* Issue: gh-102828

### CODE REVIEW

File: Lib/shutil.py [hunk 1]
```diff
@@ -692,6 +693,11 @@ def rmtree(path, ignore_errors=False, onerror=None, *, onexc=None, dir_fd=None):
     onerror is deprecated and only remains for backwards compatibility.
     If both onerror and onexc are set, onerror is ignored and onexc is used.
     """
+
+    if onerror is not None:
+        warnings.warn("onerror is deprecated, use onexc instead",
```
Comment (giampaolo) [inline Lib/shutil.py]: I would rephrase this as  `"onerror argument is deprecated, use onexc instead"` (emphasys on the fact that onerror is an argument)

Comment (iritkatriel) [inline Lib/shutil.py]:
```suggestion
        warnings.warn("onerror argument is deprecated, use onexc instead",
```

File: Lib/shutil.py [hunk 2]
```diff
@@ -692,6 +693,11 @@ def rmtree(path, ignore_errors=False, onerror=None, *, onexc=None, dir_fd=None):
     onerror is deprecated and only remains for backwards compatibility.
     If both onerror and onexc are set, onerror is ignored and onexc is used.
     """
+
+    if onerror is not None:
+        warnings.warn("onerror argument is deprecated, use onexc instead",
```
Comment (davidism) [inline Lib/shutil.py:698]: The `warn` call needs `stacklevel=2` parameter so that the warning message shows what is making the deprecated call. Werkzeug's tests started failing on the latest 3.12 alpha because of this, and it's really hard to tell what's causing it (I _think_ it's pytest) because the warning just points at this line right now instead of the caller.

Comment (iritkatriel) [reply Lib/shutil.py:698]: Thanks, will fix.
